### PR TITLE
Payment History CSV export download has correct filename (+row sorting and total row)

### PR DESCRIPTION
**Pull request by willy-b**
Payment receipt CSV links now suggest a download filename with reconciliation date in it matching UI.

~~depends on https://github.com/brave/ledger-client/pull/17/commits/d50d59e6db865a4d2da1bb4b56fe634f5f645d18~~

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist. (part of #4332)
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).

Test Plan:
1. Open/Enable Brave Payments
2. Browse pages, wait for or trigger reconciliation
3. Click 'View Payment History' link
4. Click a receipt link, e.g. `brave_ledger9-23-2016.csv`
5. Observe that a filename is suggested in download box matching link text.
6. Download CSV
7. Open from downlloads bar
8. Observe that rows are sorted alphabetically by Publisher
9. Observe that there is now a "TOTAL" row at bottom with totals for all non-publisher columns

brave_csv_export_filename_4332

### CODE REVIEW

File: app/ledger.js [hunk 1]
```diff
@@ -287,18 +286,6 @@ if (ipc) {
     if (balanceTimeoutId) clearTimeout(balanceTimeoutId)
     balanceTimeoutId = setTimeout(getBalance, 5 * msecs.second)
   })
-
-  ipc.on(messages.OPEN_LEDGER_TRANSACTION_CSV, (event, viewingIds, csvFilename) => {
```
Comment (diracdeltas) [inline app/ledger.js]: please also remove `messages.OPEN_LEDGER_TRANSACTION_CSV` and `aboutActions.receiptLinkClick`

Comment (diracdeltas) [inline app/ledger.js]: otherwise lgtm once @mrose17 releases https://github.com/brave/ledger-client/pull/17

**Comment by willy-b** [comment]
@diracdeltas, this is done (including #4332) & everything is pulled into `js/lib/ledgerExportUtil.js` so you don't need to wait on ledger-client :-)

File: js/lib/ledgerExportUtil.js [hunk 2]
```diff
@@ -0,0 +1,285 @@
+const base64Encode = require('./base64').encode
+const underscore = require('underscore')
+
+/**
+ * Generates a contribution breakdown by publisher as a CSV data URL from an array of one or more transactions
+ * @param {Object[]} transactions - array of transactions
+ */
+let transactionsToCSVDataURL = function (transactions) {
```
Comment (diracdeltas) [inline js/lib/ledgerExportUtil.js]: it would be good to add unit tests for this. (see `test/unit/lib`)

File: js/lib/ledgerExportUtil.js [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+ *
+ * @param {Object[]} transactions - array of transactions
+ * @param {string[]=} viewingIds - OPTIONAL array/string with one or more viewingIds to filter transactions by (if empty, uses all tx)
+ * @param (boolean=) addTotalRow - OPTIONAL boolean indicating whether to add a TOTALS row (defaults false)
+ **/
+let getTransactionCSVText = function (transactions, viewingIds, addTotalRow) {
+  return getTransactionCSVRows(transactions, viewingIds, addTotalRow).join('\n')
+}
+
+module.exports = {
```
Comment (diracdeltas) [inline js/lib/ledgerExportUtil.js]: for future reference, you can just use es6 object property-value shorthand here. https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand. i.e.,
```
{
  transactionsToCSVDataURL,
  getTransactionCSVText,
 ...
}

**Comment by diracdeltas** [comment]
thanks! merging for now

**Pull request merged**